### ISSUE TITLE
ansible-galaxy collection install of local collection fails if collection directory ends with '/'

### ISSUE THREAD

**Issue by antonc42** [ansible/ansible#77803]
### Summary

When trying to install a collection from a local directory, it fails if the directory has a trailing slash (/). If the trailing slash is removed, the installation succeeds.

### Issue Type

Bug Report

### Component Name

ansible-galaxy

### Ansible Version

```console
$ ansible --version
ansible [core 2.14.0.dev0]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/redacted/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible
  ansible collection location = /home/redacted/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/redacted/.virtualenvs/dev-ansible/bin/ansible
  python version = 3.10.4 (main, Apr  8 2022, 17:35:13) [GCC 9.4.0] (/home/redacted/.virtualenvs/dev-ansible/bin/python3)
  jinja version = 3.1.2
  libyaml = True
```

### Configuration

```console
$ ansible-config dump --only-changed
CONFIG_FILE() = /etc/ansible/ansible.cfg
```

### OS / Environment

Ubuntu 20.04
Python 3.10.4

### Steps to Reproduce

```yaml (paste below)
ansible-galaxy collection install -vvvvvv -f -p ~/devcollections ansible-collection-dir/
```

### Expected Results

I expected the collection to be installed.

### Actual Results

The relevant part of the output is "b'ansible-collection-dir/lugins/README.md'". As can be seen, the "plugins" directory name is being truncated by one character.

```console
ansible-galaxy [core 2.14.0.dev0]
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/redacted/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible
  ansible collection location = /home/redacted/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/redacted/.virtualenvs/dev-ansible/bin/ansible-galaxy
  python version = 3.10.4 (main, Apr  8 2022, 17:35:13) [GCC 9.4.0] (/home/redacted/.virtualenvs/dev-ansible/bin/python3)
  jinja version = 3.1.2
  libyaml = True
Using /etc/ansible/ansible.cfg as config file
Starting galaxy collection install process
[WARNING]: The specified collections path '/home/redacted/devcollections' is not part of the configured Ansible collections paths '/home/redacted/.ansible/collections:/usr/share/ansible/collections'. The installed collection will not be picked up in an Ansible run, unless
within a playbook-adjacent collections directory.
Process install dependency map
Starting collection install process
Installing 'mynamespace.mycollection:1.0.0' to '/home/redacted/devcollections/ansible_collections/mynamespace.mycollection'
Skipping 'ansible-collection-dir/.git' for collection build
Skipping 'ansible-collection-dir/galaxy.yml' for collection build
ERROR! Unexpected Exception, this is probably a bug: [Errno 2] No such file or directory: b'ansible-collection-dir/lugins/README.md'
the full traceback was:

Traceback (most recent call last):
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/cli/__init__.py", line 623, in cli_executor
    exit_code = cli.run()
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/cli/galaxy.py", line 646, in run
    return context.CLIARGS['func']()
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/cli/galaxy.py", line 102, in method_wrapper
    return wrapped_method(*args, **kwargs)
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/cli/galaxy.py", line 1300, in execute_install
    self._execute_install_collection(
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/cli/galaxy.py", line 1328, in _execute_install_collection
    install_collections(
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/galaxy/collection/__init__.py", line 719, in install_collections
    install(concrete_coll_pin, output_path, artifacts_manager)
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/galaxy/collection/__init__.py", line 1276, in install
    install_src(collection, b_artifact_path, b_collection_path, artifacts_manager)
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/galaxy/collection/__init__.py", line 1410, in install_src
    collection_output_path = _build_collection_dir(
  File "/home/redacted/.virtualenvs/dev-ansible/lib/python3.10/site-packages/ansible/galaxy/collection/__init__.py", line 1205, in _build_collection_dir
    existing_is_exec = os.stat(src_file).st_mode & stat.S_IXUSR
FileNotFoundError: [Errno 2] No such file or directory: b'ansible-collection-dir/lugins/README.md'
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by ansibot** [comment]
Files identified in the description:
* `lib/ansible/galaxy`

If these files are incorrect, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/devel/ISSUE_HELP.md)

**Comment by antonc42** [comment]
The source of the issue appears to be: https://github.com/ansible/ansible/blob/devel/lib/ansible/galaxy/collection/__init__.py#L1029

There are two possible fixes.

The first is pretty straightforward, but relies on the Python function [os.path.commonpath](https://docs.python.org/3/library/os.path.html#os.path.commonpath), which was introduced in Python 3.5.

[solution1-python3.5-above.diff.txt](https://github.com/ansible/ansible/files/8690057/solution1-python3.5-above.diff.txt)

The second is a bit more complex, but should work on older Python versions.

[solution2-below-python3.5.diff.txt](https://github.com/ansible/ansible/files/8690096/solution2-below-python3.5.diff.txt)

I do not know what the project's needs are in terms of Python version support. If someone could give me some guidance, I can submit a PR.

**Comment by s-hertel** [comment]
`os.path.commonpath` would be fine. The fix could be backported to 2.13, which requires Python 3.8+ on the controller. We could also use a [test](https://github.com/ansible/ansible/tree/devel/test/integration/targets/ansible-galaxy-collection) to prevent regression and a [changelog fragment](https://docs.ansible.com/ansible/devel/community/development_process.html#creating-a-changelog-fragment).

**Comment by antonc42** [comment]
I'm having trouble with the integration test. I tried to add something, but it doesn't seem to be working. I also had to comment out some of the already existing tests that were failing. When I run ``ansible-test integration ansible-galaxy-collection``, it pulls down a container ``quay.io/ansible/pulp-galaxy-ng`` and uses it for the tests. I don't really understand how exactly it works, but it seems that the tasks are running the ``ansible-galaxy`` commands inside the container, which would not have the patched version. Am I approaching this incorrectly?

[integration-test.patch.txt](https://github.com/ansible/ansible/files/9029290/integration-test.patch.txt)
